### PR TITLE
Update metric docs for 3.3

### PR DESCRIPTION
**Pull request by mwringe**
Metric docs updated for 3.3

### CODE REVIEW

**Comment by mwringe** [comment]
@adellape if you could take a look that would be greatly appreciated

File: install_config/cluster_metrics.adoc [hunk 1]
```diff
@@ -214,71 +219,66 @@ xref:../architecture/core_concepts/templates.adoc#architecture-core-concepts-tem
...
-between components. Because these are self-signed certificates, they are not
-automatically trusted by a web browser. Therefore, it is recommended to use
-internal certificates for anything being accessed outside of the {product-title}
-cluster, and then use the re-encrypting route to provide your own custom
-certificates. This is especially important for the Hawkular Metrics server as it
-must be accessible in a browser for the web console to function.
+The Metrics Deployer will auto-generate self-signed certificates for use between its
+components and will generate a 
+xref:../architecture/core_concepts/routes.adoc#secured-routes[re-encrypting route] to expose
+the Hawkular Metrics service. This route is what allows the console to access the Hawkular Metrics
```
Comment (adellape) [inline install_config/cluster_metrics.adoc]: s/console/web console/

File: install_config/cluster_metrics.adoc [hunk 2]
```diff
@@ -214,71 +219,66 @@ xref:../architecture/core_concepts/templates.adoc#architecture-core-concepts-tem
...
+service.
+
+In order for the browser running the console to trust the connection through this route,
```
Comment (adellape) [inline install_config/cluster_metrics.adoc]: s/console/web console/

File: install_config/cluster_metrics.adoc [hunk 3]
```diff
@@ -214,71 +219,66 @@ xref:../architecture/core_concepts/templates.adoc#architecture-core-concepts-tem
...
+$ oc secrets new metrics-deployer \
+    hawkular-metrics.pem=/home/openshift/metrics/hm.pem \
+    hawkular-metrics-ca.cert=/home/openshift/metrics/hm-ca.cert
 ----
 ====
 
-Then, use the a xref:metrics-reencrypting-route[re-encrypting route] to pass
-your custom certificates to Heapster. This allows for greater control in
-modifying the certificates in the future.
+When these secrets are provided, the deployer will uses these values to specify the
```
Comment (adellape) [inline install_config/cluster_metrics.adoc]: s/will uses/uses/

File: install_config/cluster_metrics.adoc [hunk 4]
```diff
@@ -214,71 +219,66 @@ xref:../architecture/core_concepts/templates.adoc#architecture-core-concepts-tem
...
-certificate authority certificate files:
 
-====
+[[metrics-using-secrets-autogenerated]]
+==== Using the Router's Default Certificate
+
+If the `hawkular-metrics.pem` value is not specified, the re-encrypting route will
+use the router's default certificate, which may not be trusted by browsers.
+
+A secret secred named *metrics-deployer* will still be required in this situation. This
```
Comment (adellape) [inline install_config/cluster_metrics.adoc]: "A secret secred" ?

File: install_config/cluster_metrics.adoc [hunk 5]
```diff
@@ -214,71 +219,66 @@ xref:../architecture/core_concepts/templates.adoc#architecture-core-concepts-tem
...
-Setting the value using secrets will replace the internally used certificates.
-Therefore, these certificates must be valid for both the externally used host
-names as well as the external host name. For `hawkular-metrics`, this means the
-certificate must have a value of the literal string `hawkular-metrics` as well
-as the value specified in `*HAWKULAR_METRICS_HOSTNAME*`.
 
-If you are unable to add the internal host name to your certificate, then you
-can use the xref:metrics-reencrypting-route[re-encrypting route] method.
-====
+==== Deployer Secret Options
```
Comment (adellape) [inline install_config/cluster_metrics.adoc]: Manually add a unique ID before the title, e.g.:

```
[[deployer-secret-options]]
==== Deployer Secret Options
```

**Comment by adellape** [comment]
@mwringe Some comments inline. Also, some of the new/updated links need to fixed. See the latest notes here on using xref vs link:

https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#links-hyperlinks-and-cross-references

Namely, use xrefs for linking within or between .adoc files (and don't use .html), and make sure all xrefs have an anchor ID at the end (unique IDs have been added to the top of each .adoc file already if you're just linking to the beginning of a topic).

I take it this is for OCP 3.3+? Is there a Trello card or BZ that goes with it?

Thanks!

**Comment by mwringe** [comment]
@adellape Ok, updated to fix the issue you outlined.

Yes, this is for OCP 3.3+. Are you looking for just a generic 'update to metric docs for 3.3' type of bz/trello card?

**Comment by adellape** [comment]
> Are you looking for just a generic 'update to metric docs for 3.3' type of bz/trello card?

Yeah, pretty much, if there happens to be one. If not, no worries.

File: install_config/cluster_metrics.adoc [hunk 6]
```diff
@@ -59,8 +59,8 @@ ifdef::openshift-origin[]
 ====
 If your {product-title} installation was originally performed on a version
 previous to v1.0.8, even if it has since been updated to a newer version, follow
-the instructions for node certificates outlined in Updating
-Master and Node Certificates. If the node certificate does not contain the IP
+the instructions for node certificates outlined in xref:upgrading/manual_upgrades.html#manual-updating-master-and-node-certificates[Updating
```
Comment (adellape) [inline install_config/cluster_metrics.adoc]: s/html/adoc/

File: install_config/cluster_metrics.adoc [hunk 7]
```diff
@@ -188,6 +188,11 @@ xref:../architecture/additional_concepts/storage.adoc#persistent-volume-claims[p
 volume claims] is handled by the
 xref:../install_config/cluster_metrics.adoc#metrics-deployer[Metrics Deployer].
 
+If you would like to use xref:../install_config/persistent_storage/dynamically_provisioning_pvs#install-config-persistent-storage-dynamically-provisioning-pvs[dynamically provisioned
```
Comment (adellape) [inline install_config/cluster_metrics.adoc]: Missing the .adoc

**Comment by adellape** [comment]
@mwringe Couple remaining link issues inline, then will merge.

**Comment by mwringe** [comment]
@adellape updated, hopefully I got the last remaining issues :)

File: install_config/cluster_metrics.adoc [hunk 8]
```diff
@@ -418,9 +377,15 @@ values when creating the Metrics Deployer.
...
 |The persistent volume size for each of the Cassandra nodes.
 
+|`*CASSANDRA_NODES*`
+|The number of initial Cassandra nodes to deploy
+
 |`*USE_PERSISTENT_STORAGE*`
 |Set to *true* for persistent storage; set to *false* to use non-persistent storage.
 
+|`*DYNAMICALLY_PROVISION_STORAGE*`
+|Set to *true* to allow for xref:../install_config/persistent_storage/dynamically_provisioning_pvs#install-config-persistent-storage-dynamically-provisioning-pvs[dynamically provisioned storage].
```
Comment (adellape) [inline install_config/cluster_metrics.adoc]: Needs `.adoc` before anchor.

**Comment by adellape** [comment]
@mwringe Thanks. Sorry, one last issue I missed from earlier, then I'll totally merge!

**Comment by mwringe** [comment]
@adellape Ok, I add the .adoc there.

**Comment by mwringe** [comment]
@adellape This workflow doesn't seem to be too optimal. I assume this could have gone a lot easier and faster if you could have just fixed the typo instead of having to write a message to point out where the typo is and then wait for me to fix it :) But I guess this is due to how the PR process works.

Is there another more efficient option we can use?

**Comment by adellape** [comment]
@mwringe I usually try to balance what is worth commenting on vs what can just be fixed in a follow-up. In this PR, I knew we'd be following up for style/formatting edits (I've now marked it with a `to_followup` label), but we try to be more stringent on links, specifically.

**Comment by vikram-redhat** [comment]
@adellape - please follow up early next week.

**Pull request merged**